language/move-lang/tests/move_check/typing/continue_any_type.exp

```diff
@@ -2,15 +2,7 @@ error:
 
    ┌── tests/move_check/typing/continue_any_type.move:6:17 ───
    │
- 6 │             0 + continue;
+ 6 │             foo(continue)
```

Comment:
The fact that we allow this and not `0 + continue` makes me worried. Do we not allow `x + if (cond) 1 else 2`?

In other words, after looking at this, I feel like we should be trying to flatten exp more, not adding these striations in the grammar. Thoughts?

Additionally, later I would like to be able to write `1 + loop { .. }` if we add break with a value

Reply:
No, we don't currently allow those things. I wonder if maybe you ran into problems with the grammar back when we were using lalrpop? It seems like we should revisit the expression grammar regardless.